integration-cli/docker_cli_run_test.go

```diff
@@ -1958,12 +2130,25 @@ func (s *DockerSuite) TestRunExposePort(c *check.C) {
 }
 
 func (s *DockerSuite) TestRunUnknownCommand(c *check.C) {
-	testRequires(c, NativeExecDriver, DaemonIsLinux)
+	if daemonPlatform != "windows" {
+		testRequires(c, NativeExecDriver)
+	}
 	out, _, _ := dockerCmdWithStdoutStderr(c, "create", "busybox", "/bin/nada")
 
 	cID := strings.TrimSpace(out)
 	_, _, err := dockerCmdWithError("start", cID)
-	c.Assert(err, check.NotNil)
+
+	// Windows and Linux are different here by architectural design. Linux will
+	// fail to start the container, so an error is expected. Windows will
+	// successfully start the container, and once started attempt to execute
+	// the command which will fail.
+	if daemonPlatform == "windows" {
+		// Wait for it to exit.
+		waitExited(cID, 30)
+		c.Assert(err, check.IsNil)
```

Comment:
I kind of go back to the reality statement that Windows does not behave the same as Linux and is not Linux with a different API set. They are completely different platforms. We cannot tell up front whether an application will actually start in a container without starting the container first.